homeassistant/components/lock/xiaomi_aqara.py

```diff
@@ -0,0 +1,91 @@
+"""
+Support for Xiaomi Aqara Lock.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/lock.xiaomi_aqara/
+"""
+import asyncio
+import logging
+from homeassistant.components.xiaomi_aqara import (PY_XIAOMI_GATEWAY,
+                                                   XiaomiDevice)
+from homeassistant.components.lock import LockDevice
+from homeassistant.const import (STATE_LOCKED, STATE_UNLOCKED)
+from homeassistant.helpers.event import async_call_later
+
+_LOGGER = logging.getLogger(__name__)
+
+FINGER_KEY = 'fing_verified'
+PASSWORD_KEY = 'psw_verified'
+CARD_KEY = 'card_verified'
+VERIFIED_WRONG_KEY = 'verified_wrong'
+
+ATTR_VERIFIED_WRONG_TIMES = 'verified_wrong_times'
+
+UNLOCK_MAINTAIN_TIME_S = 5
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
```

Comment:
This should be made a coroutine via `async def` to make the call to `async_call_later` in `parse_data`, which will be called during entity init, work. I assume the rest of the calls in the base entity `XiaomiDevice` can be done safely from an async context?